sklearn/model_selection/tests/test_validation.py

```diff
@@ -915,54 +916,101 @@ def test_cross_val_predict_sparse_prediction():
     assert_array_almost_equal(preds_sparse, preds)
 
 
-def check_cross_val_predict_with_method(est):
-    iris = load_iris()
-    X, y = iris.data, iris.target
-    X, y = shuffle(X, y, random_state=0)
-    classes = len(set(y))
-
+def check_cross_val_predict_with_method(est, X, y, methods):
     kfold = KFold()
 
-    methods = ['decision_function', 'predict_proba', 'predict_log_proba']
     for method in methods:
         predictions = cross_val_predict(est, X, y, method=method)
-        assert_equal(len(predictions), len(y))
 
-        expected_predictions = np.zeros([len(y), classes])
+        if isinstance(predictions, list):
+            assert_equal(len(predictions), y.shape[1])
+            for i in range(y.shape[1]):
+                assert_equal(len(predictions[i]), len(y))
+            expected_predictions = [np.zeros([len(y), len(set(y[:, i]))])
+                                    for i in range(y.shape[1])]
+        else:
+            assert_equal(len(predictions), len(y))
+            expected_predictions = np.zeros_like(predictions)
         func = getattr(est, method)
 
         # Naive loop (should be same as cross_val_predict):
         for train, test in kfold.split(X, y):
             est.fit(X[train], y[train])
-            expected_predictions[test] = func(X[test])
+            preds = func(X[test])
```

Comment:
I'm a bit confused. Doesn't this mean that sometimes we're going to have mismatched numbers of classes vis-a-vis what we try to solve with `_enforce_prediction_order`?